sdk/eventhub/event-hubs/samples/receiveEventsUsingCheckpointStore.ts

```diff
@@ -32,38 +29,35 @@ const storageConnectionString = "";
 const containerName = "";
 
 async function main() {
-  // The callback where you add your code to process incoming events
-  const processEvents = async (
-    events: ReceivedEventData[],
-    context: PartitionContext & PartitionCheckpointer
-  ) => {
-    if (events.length === 0) {
-      return;
-    }
+  const consumerClient = new EventHubConsumerClient(EventHubConsumerClient.defaultConsumerGroup, connectionString, eventHubName);
+  
+  // this client will be used by our eventhubs-checkpointstore-blob, which 
+  // persists any checkpoints from this session in Azure Storage
+  const containerClient = new ContainerClient(storageConnectionString, containerName);
 
-    for (const event of events) {
-      console.log(
-        `Received event: '${event.body}' from partition: '${context.partitionId}' and consumer group: '${context.consumerGroupName}'`
-      );
-    }
+  if (!containerClient.exists()) {
+    await containerClient.create();
+  }
 
-    // checkpoint using the last event in the batch
-    const lastEvent = events[events.length - 1];
-    await context.updateCheckpoint(lastEvent).catch((err: any) => {
-      console.log(`Error when checkpointing on partition ${context.partitionId}: `, err);
-    });
-    console.log(
-      `Successfully checkpointed event with sequence number: ${lastEvent.sequenceNumber} from partition: 'partitionContext.partitionId'`
-    );
-  };
+  const checkpointStore : CheckpointStore = new BlobPartitionManager(containerClient);
 
-  const consumerClient = new EventHubConsumerClient(connectionString, eventHubName);
-  const containerClient = new ContainerClient(storageConnectionString, containerName);
-  await containerClient.create();
   const subscription = consumerClient.subscribe(
-    EventHubConsumerClient.defaultConsumerGroupName,
-    new BlobPartitionManager(containerClient), {
-      processEvents: processEvents
+    checkpointStore, {
+      processEvent: async (event, context) => {
+        console.log(`Received event: '${event.body}' from partition: '${context.partitionId}' and consumer group: '${context.consumerGroup}'`);
+    
+        try {
+          // save a checkpoint now that we've processed this event.
+          await context.updateCheckpoint(event)
```

Comment:
In the review, we talked about the sample not updating every event.
Do we plan to make that change in a separate PR?

Reply:
Issue created: #6250 6250